### PR TITLE
Stakeholder movie invitation

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] Added a custom notification for new stakeholders invited on a movie.
- [x] Added a download link for materials with "available" state in delivery view item.
- [x] Added a download link for materials with "approved = true" in movie material item.

### CODE REVIEW

File: functions/src/stakeholder.ts [hunk 1]
```diff
@@ -150,19 +150,30 @@ export const onMovieStakeholderCreate = async (
...
+        .get()
+        .then(col => col.size);
       const orgs = await getOrgsOfMovie(movie.id);
       const notifications = orgs
         .filter(org => !!org && !!org.userIds)
         .reduce((ids: string[], { userIds }) => [...ids, ...userIds], [])
-        .map((userId: string) =>
-          prepareNotification({
+        .map((userId: string) => {
+          const message =
```
Comment (GrandSchtroumpf) [inline functions/src/stakeholder.ts]: Create an function outside of the map to do that. It's difficult to read. Especially with prettier formatting

File: libs/material/src/lib/delivery/movie-material-item/movie-material-item.component.scss [hunk 2]
```diff
@@ -48,3 +48,14 @@ p {
   color: var(--primary);
   font-weight: bolder;
 }
+
+.labo-approval {
+  position: relative;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/movie-material-item/movie-material-item.component.scss]: Why is it relative ? Do you have something with position absolute inside ?

File: libs/material/src/lib/delivery/movie-material-item/movie-material-item.component.scss [hunk 3]
```diff
@@ -48,3 +48,14 @@ p {
...
+  bottom: 70px;
+  left: 950px;
+}
+
+.done-icon {
+  position: relative;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/movie-material-item/movie-material-item.component.scss]: Same here ?

File: functions/src/delivery.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ async function notifyOnNewSignee(delivery: any, orgs: Organization[]): Promise<v
         signed delivery ${delivery.id}`,
         userId,
         path: `/layout/${delivery.movieId}/form/${delivery.id}`,
-        deliveryId: delivery.id
+        docId: delivery.id
```
[the remark this answers is not in the document]
Comment (ioaNikas) [reply functions/src/delivery.ts]: As notifications often refers to a delivery or a movie, I use a single variable for both (this variable is used to access the document and do some operations like updating the stakeholder status on a delivery or a movie)

File: functions/src/notify.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ interface PartialNotification {
   message: string;
   userId: string;
   path?: string;
-  docId?: string;
+  doc?: { id: string, type: 'movie' | 'delivery' } ,
```
Comment (GrandSchtroumpf) [inline functions/src/notify.ts]: Create a type for that : 
```typescript
interface DocId {
  id: string;
  type: 'movie' | 'delivery'
}
```

File: functions/src/stakeholder.ts [hunk 6]
```diff
@@ -22,10 +29,20 @@ export async function getOrgsOfMovie(movieId: string): Promise<Organization[]> {
   return orgs.map(doc => doc.data() as Organization);
 }
 
-export const onDeliveryStakeholderCreate = async (
+function customMessage(userId: string, snapInformations: snapObject, deliveryId?: string, movieId?: string) {
+  const documentType = (!!deliveryId) ? 'delivery' : 'movie';
+  const documentId = (!!deliveryId) ? deliveryId : movieId;
```
Comment (GrandSchtroumpf) [inline functions/src/stakeholder.ts]: You'll have a problem here is movieId is not defined.

File: functions/src/stakeholder.ts [hunk 7]
```diff
@@ -43,27 +60,26 @@ export const onDeliveryStakeholderCreate = async (
     }
 
     try {
-      const stakeholderCount = await db
+      const snapInformations = {
+        stakeholderId: newStakeholder.id,
+        orgName: newStakeholderOrg.name,
+        count: await db
```
Comment (GrandSchtroumpf) [inline functions/src/stakeholder.ts]: create a `const count = await db.collection(...).get().then(col => col.size)` above for better readibility.

File: functions/src/stakeholder.ts [hunk 8]
```diff
@@ -150,27 +165,26 @@ export const onMovieStakeholderCreate = async (
...
```
Comment (GrandSchtroumpf) [inline functions/src/stakeholder.ts]: Same here. And create a common function :
```typescript
function getCount(collection: string) {
  return db.collection(collection).get().then(col => col.size)
}
```

File: functions/src/notify.ts [hunk 9]
```diff
@@ -1,11 +1,16 @@
 import { db, serverTimestamp } from './firebase';
 
+export interface Doc {
```
Comment (GrandSchtroumpf) [inline functions/src/notify.ts]: Call it DocID instead. Because it only about the ID

**Pull request merged**